File: OPENAPI-REST-API/swagger-client/dart/lib/model/inline_response20012_filter_firewall.dart
part of swagger.api;

class InlineResponse20012FilterFirewall {
  
  List<InlineResponse20012FilterFirewallRules> rules = [];

  List<InlineResponse20012FilterFirewallFilters> filters = [];

  int scrubEnabled = null;

  InlineResponse20012FilterFirewall();

  @override
  String toString() {
    return 'InlineResponse20012FilterFirewall[rules=$rules, filters=$filters, scrubEnabled=$scrubEnabled, ]';
  }

  InlineResponse20012FilterFirewall.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    rules = InlineResponse20012FilterFirewallRules.listFromJson(json['rules']);
    filters = InlineResponse20012FilterFirewallFilters.listFromJson(json['filters']);
    scrubEnabled = json['scrub_enabled'];
  }

  Map<String, dynamic> toJson() {
    return {
      'rules': rules,
      'filters': filters,
      'scrub_enabled': scrubEnabled
     };
  }

  static List<InlineResponse20012FilterFirewall> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20012FilterFirewall>() : json.map((value) => new InlineResponse20012FilterFirewall.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20012FilterFirewall> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, InlineResponse20012FilterFirewall>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new InlineResponse20012FilterFirewall.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/InlineResponse20012FilterFirewall.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse20012FilterFirewall {
    /// <summary>
    /// Gets or Sets Rules
    /// </summary>
    [DataMember(Name="rules", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "rules")]
    public List<InlineResponse20012FilterFirewallRules> Rules { get; set; }

    /// <summary>
    /// Gets or Sets Filters
    /// </summary>
    [DataMember(Name="filters", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "filters")]
    public List<InlineResponse20012FilterFirewallFilters> Filters { get; set; }

    /// <summary>
    /// Gets or Sets ScrubEnabled
    /// </summary>
    [DataMember(Name="scrub_enabled", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_enabled")]
    public int? ScrubEnabled { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse20012FilterFirewall {\n");
      sb.Append("  Rules: ").Append(Rules).Append("\n");
      sb.Append("  Filters: ").Append(Filters).Append("\n");
      sb.Append("  ScrubEnabled: ").Append(ScrubEnabled).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
